functions/venue.js

```diff
@@ -410,7 +410,7 @@ exports.toggleDustStorm = functions.https.onCall(async (_data, context) => {
 });
 
 exports.updateVenue = functions.https.onCall(async (data, context) => {
-  const venueId = getVenueId(data.name);
+  const venueId = data.id || getVenueId(data.name);
```

Comment:
[Future Cleanup] We should remove `getVenueId(data.name)` from this entirely (and any other backend calls like this).

As per my comments in https://github.com/sparkletown/sparkle/pull/1228#issuecomment-789199882, we shouldn't use a 'name' as an 'id' (even in this case where it's passed through a function to derive it)

> The fact that the name (which IMO should only be a human facing label) is used to derive the id forever seems super broken to me. It should be used to optionally derive the id once in the beginning, but then we should just be using the id directly from that point onwards.